nemo/collections/asr/parts/utils/diarization_utils.py

```diff
@@ -527,71 +221,52 @@ def _get_speech_labels_from_decoded_prediction(self, input_word_ts):
                 count -= 1
         return word_ts
 
-    def get_word_ts_from_spaces(self, char_ts, _spaces_in_sec, end_stamp):
-        """
-        Get word-timestamps from the spaces in the decoded prediction.
-
-        Args:
-            char_ts (list):
-                The time-stamps for each character.
-            _spaces_in_sec (list):
-                List contains the start and the end time of each space.
-            end_stamp (float):
-                The end time of the session in sec.
-
-        Return:
-            word_timestamps (list):
-                List of the timestamps for the resulting words.
-        """
-        start_stamp_in_sec = round(char_ts[0] * self.params['time_stride'] - self.asr_delay_sec, 2)
-        end_stamp_in_sec = round(end_stamp * self.params['time_stride'] - self.asr_delay_sec, 2)
-        word_timetamps_middle = [
-            [
-                round(_spaces_in_sec[k][1] - self.asr_delay_sec, 2),
-                round(_spaces_in_sec[k + 1][0] - self.asr_delay_sec, 2),
-            ]
-            for k in range(len(_spaces_in_sec) - 1)
-        ]
-        word_timestamps = (
-            [[start_stamp_in_sec, round(_spaces_in_sec[0][0] - self.asr_delay_sec, 2)]]
-            + word_timetamps_middle
-            + [[round(_spaces_in_sec[-1][1] - self.asr_delay_sec, 2), end_stamp_in_sec]]
-        )
-        return word_timestamps
-
     def run_diarization(
-        self, diar_model_config, words_and_timestamps,
+        self, diar_model_config, word_timestamps,
     ):
         """
-        Run the diarization process using the given VAD timestamp (oracle_manifest).
+        Launch the diarization process using the given VAD timestamp (oracle_manifest).
 
         Args:
             word_and_timestamps (list):
-                List contains words and word-timestamps
+                List containing words and word timestamps
+
+        Returns:
+            diar_hyp (dict):
+                A dictionary containing rttm results which are indexed by a unique ID.
+            score Tuple[pyannote object, dict]:
+                A tuple containing pyannote metric instance and mapping dictionary between
+                speakers in hypotheses and speakers in reference RTTM files.
         """
 
         if diar_model_config.diarizer.asr.parameters.asr_based_vad:
-            self.save_VAD_labels_list(words_and_timestamps, self.audio_file_list)
+            self.save_VAD_labels_list(word_timestamps)
             oracle_manifest = os.path.join(self.root_path, 'asr_vad_manifest.json')
             oracle_manifest = write_rttm2manifest(self.VAD_RTTM_MAP, oracle_manifest)
             diar_model_config.diarizer.vad.model_path = None
             diar_model_config.diarizer.vad.external_vad_manifest = oracle_manifest
 
         oracle_model = ClusteringDiarizer(cfg=diar_model_config)
```

Comment:
Can we rename this to diar_model, since it's not oracle_model always? Helps for clear understanding

Reply:
Changed to diar_model